lib/tasks/tag_tasks.rake

```diff
@@ -36,16 +36,33 @@ namespace :Tag do
 
   desc "Delete unused tags"
   task(delete_unused: :environment) do
+    start = Time.current
     deleted_names = []
-    Tag.where(canonical: false, merger_id: nil, taggings_count_cache: 0).each do |t|
-      if t.taggings.count.zero? && t.child_taggings.count.zero? && t.set_taggings.count.zero?
+    Tag.where(canonical: false, merger_id: nil, taggings_count_cache: 0).find_in_batches do |batch|
+      batch.each do |t|
+        next unless t.taggings.count.zero?
+        next unless t.child_taggings.count.zero?
+        next unless t.set_taggings.count.zero?
+
         deleted_names << t.name
-        t.destroy
+        begin
+          deleted_names << t.name
```

Comment:
This pushes each tag into the array (`deleted_names << t.name`) at least twice. I think you'll want to remove the second statement here, otherwise it will also push it into the array again for each retry.